configs/components/curl.rb

```diff
@@ -9,17 +9,16 @@
     pkg.apply_patch 'resources/patches/curl/curl-7.55.1-aix-poll.patch'
   end
 
-  pkg.build_requires "openssl"
+  pkg.build_requires "openssl-#{settings[:runtime_project]}"
```

Comment:
I'd definitely prefer to have some branching in the openssl component to see if we can combine these into one component, or figure out where we actually need differences vs where the differences are just due to drift. I won't block on this for this PR, but I consider having two components for the same version of the same software sort of a last resort.